src/component/Timer/index.jsx

```diff
@@ -22,11 +31,43 @@ const Timer = () => {
 
     const reset = () => {
       timeOverSoundAudio.play();
+      const showNotification = () => {
+        let text = null;
+        const img = 'https://source.unsplash.com/1600x900/?nature,water;';
+        if (counter.current === 6) {
+          text = `Great job! Take a long break. You have ${PERIODS.longBrk.secs} minutes.`;
+          (() => new Notification('Pomodoro', {
+            body: text,
+            icon: img
+          }))();
+        } else if (currentPeriod.id === 'short-brk' || currentPeriod.id === 'long-brk') {
+          text = `Time to get back to work! Your next break starts in ${PERIODS.work.secs} minutes.`;
+          (() => new Notification('Pomodoro', {
+            body: text,
+            icon: img
+          }))();
+        } else if (currentPeriod.id === 'work') {
+          text = `Nice work! Take a short break. You have ${PERIODS.shortBrk.secs} minutes.`;
+          (() => new Notification('Pomodoro', {
+            body: text,
+            icon: img
+          }))();
+        }
+      };
+      if (Notification.permission === 'granted') {
+        showNotification();
+      } else if (Notification.permission !== 'denied') {
+        Notification.requestPermission().then((permission) => {
+          if (permission === 'granted') {
+            showNotification();
+          }
+        });
+      }
       setIsRunning(false);
       counter.current += 1;
       const nextPeriod = determineNextPeriod(currentPeriod, counter.current);
       if (counter.current === 7) {
-        counter.current = 0;
+        counter.current = 0;// the counter need to be -1 instead of 0 becuase there is a bug
```

Comment:
Can you explain why it should be `-1`?

Reply:
because there is a bug that shows to me at the second 7 periods and it continues to the coming Pomodoro periods. at the 5th short break, it shows a long break instead of the work period. The counter starts from 0 but after the first 7 periods it starts from 1  so this problem happens and I solve it by changing the current counter to (-1).